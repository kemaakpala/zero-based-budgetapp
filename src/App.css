.App {
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  /* grid-template-rows: auto auto auto 40px;
   */
  grid-auto-rows: auto;
  grid-gap: 8px;
  height: 100%;
}

.App > *:not(aside) {
  display: flex;
  flex-direction: column;
}

/* .App > header {
  flex-direction: row;
  justify-content: space-between;
} */
.form-control {
  /* styles for the form-control block */
}

.form-control--error {
  /* styles for form-control elements with the error modifier */
}

.form-control--disabled {
  /* styles for form-control elements with the disabled modifier */
}
