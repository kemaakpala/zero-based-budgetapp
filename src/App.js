import "./App.css";

function App() {
  return (
    <div className="App">
      <header className="App-header">Zero Based Budgeting App</header>
      <main>
        <form className="form">
          <div className="income-container">
            <div className="total-income-hero__text">£ 0.00</div>
            <div className="income-fields">
              <div className="primary-income-fields">
                <label
                  className="primary-income-fields__label"
                  for="primaryIncome"
                >
                  Salary 1
                </label>
                <input
                  id="primaryIncome"
                  className="form-control primary-income-fields__input"
                  type="text"
                  name="primary-income"
                  value=""
                />
              </div>
              <div className="secondary-income-fields">
                <label
                  className="secondary-income-fields__label"
                  for="secondaryIncome1"
                >
                  Salary 2
                </label>
                <input
                  className="form-control secondary-income-fields__input"
                  id="secondaryIncome1"
                  type="text"
                  name="secondary-income-1"
                  value=""
                />
              </div>
            </div>
          </div>
          <div className="group-container">
            <div className="group-header">
              <h3>Savings</h3>
            </div>
            <div className="group-item">
              <div className="group-item-fields">
                <label className="group-item-fields__label">Emergency Fund: </label>
                <input className="form-control group-item-fields__input" type="text" name="savings_emergency_fund" value="" />
              </div>
              <div className="group-item-status">
                <h4 className="group-item-status__header">Planned</h4>
                <p className="group-item-status__text">£0.00</p>
              </div>
              <div className="group-item-status">
                <h4 className="group-item-status__header">Remaining</h4>
                <p className="group-item-status__text">£0.00</p>
              </div>
              <div className="group-item-action">
                <button className="group-item-action__button">Delete Item</button>
              </div>
            </div>
            <div className="group-actions">
              <button className="form-control group-actions__button">Add New Group Item</button>
              <button className="form-control group-actions__button">Delete Group</button>
            </div>
          </div>
          <div className="group-container">
            <div className="group-header">
              <h3>Housing</h3>
            </div>
            <div className="group-item">
              <div className="group-item-fields">
                <label className="group-item-fields__label">Rent / Mortgage: </label>
                <input className="form-control group-item-fields__input" type="text" name="housing_rent_mortgage" value="" />
              </div>
              <div className="group-item-status">
                <h4 className="group-item-status__header">Planned</h4>
                <p className="group-item-status__text">£0.00</p>
              </div>
              <div className="group-item-status">
                <h4 className="group-item-status__header">Remaining</h4>
                <p className="group-item-status__text">£0.00</p>
              </div>
              <div className="group-item-action">
                <button className="group-item-action__button">Delete Item</button>
              </div>
            </div>
            <div className="group-actions">
              <button className="form-control group-actions__button">Add New Group Item</button>
              <button className="form-control group-actions__button">Delete Group</button>
            </div>
          </div>
          <div className="group-container">
            <div className="group-header">
              <h3>Transport</h3>
            </div>
            <div className="group-item">
              <div className="group-item-fields">
                <label className="group-item-fields__label">TFL: </label>
                <input className="form-control group-item-fields__input" type="text" name="transport_tfl" value="" />
              </div>
              <div className="group-item-status">
                <h4 className="group-item-status__header">Planned</h4>
                <p className="group-item-status__text">£0.00</p>
              </div>
              <div className="group-item-status">
                <h4 className="group-item-status__header">Remaining</h4>
                <p className="group-item-status__text">£0.00</p>
              </div>
              <div className="group-item-action">
                <button className="group-item-action__button">Delete Item</button>
              </div>
            </div>
            <div className="group-actions">
              <button className="form-control group-actions__button">Add New Group Item</button>
              <button className="form-control group-actions__button">Delete Group</button>
            </div>
          </div>
          <div className="form-actions">
            <button className="form-control form-actions__button">Add New Group</button>
            <button classNam="form-control form-actions__button">Submit Budget</button>
          </div>
        </form>
      </main>
      <footer>Budget Footer</footer>
    </div>
  );
}

export default App;
