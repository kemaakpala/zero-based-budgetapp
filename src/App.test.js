import { render, screen } from "@testing-library/react";
import { RouterProvider } from "react-router-dom";
import browserRouter from "./Routes/router";
import App from "./App";

jest.mock("@fortawesome/react-fontawesome", () => ({
  FontAwesomeIcon: ({ icon, ...props }) => (
    <svg {...props}>
      <title>{props.title}</title>
      <path d="" />
    </svg>
  ),
}));

test("renders learn react link", () => {
  const router = browserRouter({
    path: "/",
    element: <App />,
  });
  const {
    container: { firstChild },
  } = render(<RouterProvider router={router} />);
  expect(firstChild).toMatchSnapshot();
});
