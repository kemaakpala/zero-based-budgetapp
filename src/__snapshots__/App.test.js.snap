// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <nav
      class="menu"
    >
      <h1>
        Every Pound (£)
      </h1>
      <div
        class="user-profile-container"
      >
        <svg
          title="user profile image"
        >
          <title>
            user profile image
          </title>
          <path
            d=""
          />
        </svg>
      </div>
      <div
        class="budget-summary-container"
      >
        <svg
          title="budget summary"
        >
          <title>
            budget summary
          </title>
          <path
            d=""
          />
        </svg>
      </div>
      <div
        class="budget-search-container"
      >
        <svg
          title="budget search"
        >
          <title>
            budget search
          </title>
          <path
            d=""
          />
        </svg>
      </div>
    </nav>
    <hr
      class="divider"
    />
  </header>
  <main
    class="App-main"
  />
  <footer
    class="App-footer"
  >
    <hr />
    <p>
      © 
      2025
    </p>
  </footer>
</div>
`;
